tests/parser/functions/test_convert_to_bytes32.py

```diff
@@ -80,3 +80,64 @@ def testConvertBytes32(flag: bool) -> bytes32:
     trueBytes = c.testConvertBytes32(True)
     assert trueBytes[31:32] == b'\x01'
     assert len(trueBytes) == 32
+
+
+def test_convert_from_decimal(get_contract_with_gas_estimation):
+    code = """
+bar: decimal
+nar: decimal
+mar: decimal
+
+@public
+def foo() -> bytes32:
+    return convert(0.0, bytes32)
+
+@public
+def foobar() -> bytes32:
+    self.bar = 0.0
+    return convert(self.bar, bytes32)
+
+@public
+def hoo() -> bytes32:
+    return convert(MIN_DECIMAL, bytes32)
+
+@public
+def hoonar() -> bytes32:
+    self.nar = MIN_DECIMAL
+    return convert(self.nar, bytes32)
+
+@public
+def goo() -> bytes32:
+    return convert(MAX_DECIMAL, bytes32)
+
+@public
+def goomar() -> bytes32:
+    self.mar = MAX_DECIMAL
+    return convert(self.mar, bytes32)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+
+    fooVal = c.foo()
+    foobarVal = c.foobar()
+    assert fooVal == (b"\x00" * 32)
+    assert len(fooVal) == 32
+    assert foobarVal == (b"\x00" * 32)
+    assert len(foobarVal) == 32
+    assert fooVal == foobarVal
+
+    hooVal = c.hoo()
+    hoonarVal = c.hoonar()
+    assert hooVal[0:11] == (b"\xff" * 11)
```

Comment:
Definitely need some comments here and similar other lines in this test to describe why certain bit patters are set.

Will docs be added to describe what the conversion will look like?

Reply:
@fubuloubu great point, I'll add some comments noting the decimal precision and why these tests are like this + also will add more tests

I'll also add to the VIP at #1093 and to PR #1501 to describe in a bit more detail what this specific conversion looks like, as the decimal precision part is a bit unintuitive 😅 

Will take care of this tonight 👍

Reply:
You the man!

Reply:
@fubuloubu updated! 😄